Migrate GitHub Actions updates from TSCCR to Dependabot

NodyHub opened the pull request:
This repository previously had GitHub Actions updates managed through the [TSCCR automation system](https://github.com/hashicorp/security-tsccr/blob/main/tsccr-helper/help/automation.md). As TSCCR is no longer being maintained, responsibility for managing GitHub Actions updates now falls to the repository's owners.

This PR enhances your Dependabot configuration by adding the GitHub Actions ecosystem. The configuration is aligned with the previous TSCCR automation approach and will ensure your GitHub Actions continue to receive important updates.

Please note that this is a one-time configuration change. After merging, Dependabot will automatically manage GitHub Actions updates according to the specified schedule.

For additional information, please refer to [Memo SEC-032](https://docs.google.com/document/d/1bvwk8yNEakgPWG5l7pKPpwwrgXnNLdMIDpMZ-TSEJo4). If you have any questions, the [#team-prodsec](https://hashicorp.enterprise.slack.com/archives/C010VJT0FRP) team in Slack would be happy to help.

huikang commented:
Please fix the failed CI tests. Thanks.

huikang commented:
> > Please fix the failed CI tests. Thanks.
> 
> It looks like the test are already broke before the PR 😬 I have not touched any line. I will check if I can prfioritize fixing the test suite, but do not think that i am the right person to address this problem 😬

Yeah, I just fixed that. Could you try rebasing the PR against the latest main. Sorry about that.

NodyHub commented:
> > > Please fix the failed CI tests. Thanks.
> > 
> > 
> > It looks like the test are already broke before the PR 😬 I have not touched any line. I will check if I can prfioritize fixing the test suite, but do not think that i am the right person to address this problem 😬
> 
> Yeah, I just fixed that. Could you try rebasing the PR against the latest main. Sorry about that.

Thank you a lot @huikang !! I rebased the pr and added as well the label `pr/no-changelog`.
